content/3_Silver/Binary_Search.mdx

```diff
@@ -408,9 +408,17 @@ as possible after $k$ operations.
 odd.
 
 <Spoiler title="Solution">
+To get the current median, we first sort the array in ascending order.
 
-We first sort the array in ascending order. Then, we binary search for the
-maximum possible median. We know that the number of operations required to raise
+Now, notice that to increase the current value to a value, say $x$.
+All values currently greater than or equal to the median, must remain greater than or equal to the median. i.e
+
+If after sorting an array we have the array
+$[1,1,2,3,4,4,5,5,6,8,8]$. The current median is $4$, to increase the median to $6$, we have to increase the current median by $2$ and we also have to increase the $5$'s to $6$.
+
+Following this idea, to increase the median to a value $x$, we will need to increase the value of $array[i]$ (for $i$ starting from  $\frac{n + 1}{2}$ (the  position of the median) down to $n$), if it is less than the value $x$.
```

Comment:
```suggestion
Following this idea, to increase the median to $x$, we need to increase the value of $array[i]$ (for $i$ starting from $\frac{n + 1}{2}$ (the position of the median) down to $n$), if it is less than the value $x$.
```
> for $i$ starting from $\frac{n + 1}{2}$ (the position of the median) down to $n$

what does this mean?

Reply:
it means we are only looking at values of the array whose position is in the range (n+1/2) to n inclusive.
and we are starting from n+1/2 because it is the current position of the median